pkg/controller/installation/typha_autoscaler_test.go

```diff
@@ -117,10 +144,10 @@ func verifyTyphaReplicas(c client.Client, expectedReplicas int) {
 	Eventually(func() int32 {
 		err := test.GetResource(c, typha)
 		Expect(err).To(BeNil())
-		// Just return an invalid number that will never match an expected replica count.
+		// Replicas defaults to 1:  https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#replicas
 		if typha.Spec.Replicas == nil {
-			return -1
+			return 1
```

Comment:
In the case where replicas should be 1, do we still expect our autoscaling to set 1 as the value? If so I think we should keep this as it was.

Reply:
Our logic for updating typha replicas will not update the deployment if the existing replicas is nil (defaults to 1) and the expected replicas is 1. What we can do is to change that logic so that it always updates the deployment if replicas is nil.